### ISSUE TITLE
No Session found for current thread

### ISSUE THREAD

**Issue by sudosandwich3** [alexkramer/grails-hibernate-filter#6]
HibernateFilterInterceptor assumes a session exists in the current thread.  In the case of error or the user not being authorized,  this will not defined and an exception is thrown.   There probably needs to be a check here:
```groovy
boolean before() {
    def session = sessionFactory.currentSession
    DefaultHibernateFiltersHolder.defaultFilters.each {
        session.enableFilter it
    }
    true
}
```

**Comment by vsachinv** [comment]
@sudosandwich3 I have also faced the same issue. Did you find any fix or fixed using inline hibernate-plugin?

**Comment by alexkramer** [comment]
Can you provide either written steps to reproduce this or an actual test case? I am having trouble understanding how to get into this scenario.

As for a workaround, the interceptor is a spring bean. Did you try to define your own implementation of the interceptor that overrides the interceptor included by this plugin?

**Comment by vsachinv** [comment]
@alexkramer I have created a custom hibernate interceptor filter bean to override now.   

> class HibernateFilterInterceptor implements Interceptor {

    private static Logger log = LoggerFactory.getLogger(HibernateFilterInterceptor)

    SessionFactory sessionFactory

    RxHibernateFilterInterceptor() {
        matchAll()
                .excludes(uri: "/static/**").excludes(uri: "/img/**")
                .excludes(uri: "/js/*").excludes(uri: "/css/**")
                .excludes(uri: "/assets/**").excludes(uri: "/**.**")
                .excludes(uri: "/images/**")
                .excludes(uri: "/console/**")
    }

    boolean before() {
        Session session
        try {
            session = sessionFactory.currentSession
        } catch (HibernateException ex) {
            log.debug("No Hibernate session found for HibernateFilterInterceptor")
            log.debug(ex.message)
            session = null
        }
        if (session) {
            DefaultHibernateFiltersHolder.defaultFilters.each {
                session.enableFilter it
            }
        }
        true
    }
}

**Comment by alexkramer** [comment]
The fix for this should be merged and published to the bintray repo as 0.5.3.
